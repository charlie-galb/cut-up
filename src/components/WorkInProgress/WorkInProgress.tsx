import React from 'react'

import { DragOverlay, useDroppable } from "@dnd-kit/core"
import { SortableContext, rectSortingStrategy} from "@dnd-kit/sortable"

import { TextSnippet } from "../TextSnippet/TextSnippet"
import { SortableTextSnippet } from "../TextSnippet/SortableTextSnippet"
import { chunkContainer } from "../../types/chunkContainer"

interface Props {
    chunkContainers: {
        [key: string]: chunkContainer
      }
    lineOrder: string[]
    addLine: (arg: void) => void
    wordChunks: {
        [key: string]: string
      }
    activeId: string
}

export const WorkInProgress = (props: Props) => {

    const { chunkContainers, lineOrder, addLine, wordChunks, activeId } = props
    const { id, title, nestedChunkIDs } = chunkContainers["chunk-container-2"]
    const { setNodeRef } = useDroppable({ id });
    const handleLineAdding = () => {
        addLine()
    }

    return (
        // <SortableContext id={id} items={nestedChunkIDs} strategy={horizontalListSortingStrategy}>
        //     <div className="wip-container">
        //         <div className="lines-container">
        //             {lineOrder?.map((lineId, i) => {
        //                 return (
        //                     <Line key={i} wordChunks={wordChunks} chunkContainer={chunkContainers[lineId]}/>
        //                 )
        //             })}
        //         </div>
        //         <button data-testid="add-line-btn" className="add-line-btn" onClick={handleLineAdding} >Add line</button>
        //     </div>
        // </SortableContext>
        <div className="wip-container"> 
            <SortableContext id={id} items={nestedChunkIDs} strategy={rectSortingStrategy}>
                <div className="wip-snippets" data-testid="unused-snippets" ref={setNodeRef}>
                    {nestedChunkIDs?.map((ID, i) => {
                        return (
                            <SortableTextSnippet data-testid="snippet" key={ID} id={ID} index={i} text={wordChunks[ID]}/>
                        )
                    })}
                </div>
            </SortableContext>
            <DragOverlay dropAnimation={null}>
                {activeId ? <TextSnippet text={wordChunks[activeId]}/> : null}
            </DragOverlay>
        </div>
    )
}
