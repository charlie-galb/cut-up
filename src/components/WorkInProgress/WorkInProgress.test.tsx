import React from "react"
import { render } from "@testing-library/react"
import { DndContext } from '@dnd-kit/core'
 

import { WorkInProgress } from "./WorkInProgress"

import { mockChunks, 
        mockContainers,
        mockLineOrder } from "../../utils/mockData"

describe("WorkInProgress", () => {
    test("It displays the correct number of text snippets", () => {
        const wip = render(
            <DndContext onDragEnd={() => {}}>
                <WorkInProgress 
                    activeId={""} 
                    lineOrder={mockLineOrder}
                    wordChunks={mockChunks} 
                    chunkContainers={mockContainers}
                    setChunkContainers={jest.fn()}
                    setLineOrder={jest.fn()}/>
            </DndContext>
        )
        expect(wip).toMatchSnapshot()
    })
})