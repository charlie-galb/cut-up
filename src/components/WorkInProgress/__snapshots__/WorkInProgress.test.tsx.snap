// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkInProgress It displays the correct number of text snippets 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wip-container"
      >
        <div
          class="lines-container"
        >
          <div>
            <div
              class="line"
              data-testid="pasteboard"
            >
              <div>
                <button
                  aria-describedby="DndDescribedBy-0"
                  aria-roledescription="sortable"
                  class="text-snippet"
                  role="button"
                  tabindex="0"
                  text="one"
                >
                  one
                </button>
              </div>
              <div>
                <button
                  aria-describedby="DndDescribedBy-0"
                  aria-roledescription="sortable"
                  class="text-snippet"
                  role="button"
                  tabindex="0"
                  text="two"
                >
                  two
                </button>
              </div>
            </div>
          </div>
          <div>
            <div
              class="line"
              data-testid="Line-1"
            >
              <div>
                <button
                  aria-describedby="DndDescribedBy-0"
                  aria-roledescription="sortable"
                  class="text-snippet"
                  role="button"
                  tabindex="0"
                  text="one"
                >
                  one
                </button>
              </div>
              <div>
                <button
                  aria-describedby="DndDescribedBy-0"
                  aria-roledescription="sortable"
                  class="text-snippet"
                  role="button"
                  tabindex="0"
                  text="two"
                >
                  two
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <button
            class="btn add-line-btn"
            data-testid="add-line-btn"
          >
            +
          </button>
        </div>
        <div>
          <button
            class="btn add-line-btn"
            data-testid="add-line-btn"
          >
            -
          </button>
        </div>
      </div>
    </div>
    <div
      id="DndDescribedBy-0"
      style="display: none;"
    >
      
    To pick up a draggable item, press the space bar.
    While dragging, use the arrow keys to move the item.
    Press space again to drop the item in its new position, or press escape to cancel.
  
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="DndLiveRegion-0"
      role="status"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%); white-space: nowrap;"
    />
  </body>,
  "container": <div>
    <div
      class="wip-container"
    >
      <div
        class="lines-container"
      >
        <div>
          <div
            class="line"
            data-testid="pasteboard"
          >
            <div>
              <button
                aria-describedby="DndDescribedBy-0"
                aria-roledescription="sortable"
                class="text-snippet"
                role="button"
                tabindex="0"
                text="one"
              >
                one
              </button>
            </div>
            <div>
              <button
                aria-describedby="DndDescribedBy-0"
                aria-roledescription="sortable"
                class="text-snippet"
                role="button"
                tabindex="0"
                text="two"
              >
                two
              </button>
            </div>
          </div>
        </div>
        <div>
          <div
            class="line"
            data-testid="Line-1"
          >
            <div>
              <button
                aria-describedby="DndDescribedBy-0"
                aria-roledescription="sortable"
                class="text-snippet"
                role="button"
                tabindex="0"
                text="one"
              >
                one
              </button>
            </div>
            <div>
              <button
                aria-describedby="DndDescribedBy-0"
                aria-roledescription="sortable"
                class="text-snippet"
                role="button"
                tabindex="0"
                text="two"
              >
                two
              </button>
            </div>
          </div>
        </div>
      </div>
      <div>
        <button
          class="btn add-line-btn"
          data-testid="add-line-btn"
        >
          +
        </button>
      </div>
      <div>
        <button
          class="btn add-line-btn"
          data-testid="add-line-btn"
        >
          -
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
